DiscTests/Networking/Authorization/GetAccessTokenSpec.swift

```diff
@@ -7,42 +7,237 @@ import Result
+                    self.stub(matcher, builder: builder)
+                    
+                    var responseValue: AccessToken?
+                    var responseError: NSError?
+                    
+                    APIClient.getAccessToken(clientId: clientId, clientSecret: clientSecret, code: code) { result in
+                        responseValue = result.value
+                        responseError = result.error
+                    }
+                    
+                    expect(responseValue).toEventually(equal(token))
+                    expect(responseError).toEventually(beNil())
+                }
+            }
+            
+            context("with an OAuth provider") {
+                var url: String!
+                
+                var providerAccessToken: String!
+                var providerTokenSecret: String!
                 
-                let matcher = api(.POST, url, body: requestBody)
-                let builder = json(responseBody)
-                self.stub(matcher, builder: builder)
+                var providerAuthCode: String!
+                var providerRedirectUri: String!
                 
-                var responseValue: AccessToken?
-                var responseError: NSError?
+                var provider: Provider!
+                var scopes: [Scope]!
                 
-                APIClient.getAccessToken(clientId: clientId, clientSecret: clientSecret, code: code) { result in
-                    responseValue = result.value
-                    responseError = result.error
+                beforeEach {
+                    url = "https://passport.thegrid.io/api/auth/login"
+                    
+                    providerAccessToken = "provider access token"
+                    providerTokenSecret = "provider token secret"
+                    
+                    providerAuthCode = "provider auth code"
+                    providerRedirectUri = "provider redirect URI"
+                    
+                    provider = .Twitter
+                    scopes = [ .ContentManagement, .UpdateProfile ]
                 }
                 
-                expect(responseValue).toEventually(equal(token))
-                expect(responseError).toEventually(beNil())
+                context("access token") {
+                    context("without scopes") {
+                        it("should result in an access token") {
+                            let requestBody = [
+                                "provider": provider.rawValue,
+                                "token": providerAccessToken,
+                            ]
+                            
+                            let matcher = api(.POST, url, body: requestBody)
+                            let builder = json(responseBody)
+                            self.stub(matcher, builder: builder)
+                            
+                            var responseValue: AccessToken?
```

Comment:
Curious, if these optionals were moved to `beforeEach` would there be less duplication with the same result?